### PR TITLE
Add newline-less `Stdout.write` & `Stderr.write` to `cli-platform`

### LINKED ISSUE #4120: `cli-platform` `Stdout.line` alternative without newline suffix?
The `line` function in `examples/interactive/cli-platform/Stdout` is the only way to write to stdout, and it corresponds to `println!(...)` in Rust. Can we please add a second exposed function in `Stdout` that doesn't append a newline every time? I would use this immediately in https://github.com/JanCVanB/rocbox, but I also expect that other people's CLIs will need return un-newlined output, too.

I don't know what to call it, though... `print` feels lame... `write`? Perhaps the pair should be `write` & `writeLine`?

### PR DESCRIPTION
**Pull request by JanCVanB**
Closes #4120

### CODE REVIEW

**Comment by JanCVanB** [comment]
I'm only a Rustling, so let me know if the flushing looks right. [`eprint!` doesn't seem to need it](https://doc.rust-lang.org/std/macro.eprint.html), while [`print!` does](https://doc.rust-lang.org/std/macro.print.html).

**Review by Anton-4 — DISMISSED** [review]
Thanks @JanCVanB!
I did notice two `UNUSED DEFINITION` warnings in the `cli_run` tests

**Comment by ayazhafiz** [comment]
I am pretty sure `eprint` will need a flush as well - probably just an omission not to mention it in the rust docs

**Comment by JanCVanB** [comment]
@Anton-4 Whoops, fixing now

**Comment by JanCVanB** [comment]
@ayazhafiz stderr might be unbuffered in Rust: https://github.com/rust-lang/rust/issues/64413 ... is there any easy way for me to check?

**Comment by ayazhafiz** [comment]
Writing to `stderr` does not flush by default in Rust, so it just goes to the underlying OS standard error buffer immediately. When that flushes is OS-dependent. I think we should either always flush or never flush explicitly here; given the relative low cost of flushing, I am in favor of always flushing after writes to both stdout and stderr. What do you think @JanCVanB ?

**Comment by JanCVanB** [comment]
@ayazhafiz I agree! Thank you for finding/communicating that nuance - I'll add an eprint flush

File: examples/cli/cli-platform/src/lib.rs [hunk 1]
```diff
@@ -276,12 +277,26 @@ pub extern "C" fn roc_fx_stdoutLine(line: &RocStr) {
...
 pub extern "C" fn roc_fx_stderrLine(line: &RocStr) {
     let string = line.as_str();
     eprintln!("{}", string);
 }
 
+#[no_mangle]
+pub extern "C" fn roc_fx_stderrWrite(text: &RocStr) {
+    let string = text.as_str();
+    eprint!("{}", string);
+    io::stdout().flush().unwrap();
```
Comment (ayazhafiz) [inline examples/cli/cli-platform/src/lib.rs:297]:
```suggestion
    io::stderr().flush().unwrap();
```

Comment (JanCVanB) [reply examples/cli/cli-platform/src/lib.rs:297]: Whoops, thanks

**Comment by JanCVanB** [comment]
I've suggested that Rust update their `eprint!` docs: https://github.com/rust-lang/www.rust-lang.org/issues/1721

**Review by ayazhafiz — APPROVED** [review]

**Pull request merged**